sequencer/src/api/endpoints.rs

```diff
@@ -121,3 +124,52 @@ where
 
     Ok(api)
 }
+
+pub(super) fn state_signature<N, S>() -> anyhow::Result<Api<S, Error>>
+where
+    N: network::Type,
+    S: 'static + Send + Sync + ReadState,
+    S::State: Send + Sync + StateSignatureDataSource<N>,
+{
+    let toml = toml::from_str::<toml::Value>(include_str!("../../api/state_signature.toml"))?;
+    let mut api = Api::<S, Error>::new(toml)?;
+
+    api.get("getstatesignature", |req, state| {
+        async move {
+            let height = req
+                .integer_param("height")
+                .map_err(|err| Error::internal(err.to_string()))?;
+            state
+                .handle()
+                .get_state_signature(height)
+                .ok_or(Error::internal("Signature not found."))
```

Comment:
Probably not great to return an internal error here, we should return something with a 404 error code. You should be able to do `Error::catch_all(StatusCode::NotFound, "signature not found")`

Reply:
using this in [f7033f7](https://github.com/EspressoSystems/espresso-sequencer/pull/879/commits/f7033f7222880be8e53eeee798c12ca2691f496c)